library/alloc/src/collections/btree/set.rs

```diff
@@ -696,53 +688,46 @@ impl<T, A: Allocator + Clone> BTreeSet<T, A> {
         if self.len() > other.len() {
             return false;
         }
-        let (self_min, self_max) =
-            if let (Some(self_min), Some(self_max)) = (self.first(), self.last()) {
-                (self_min, self_max)
-            } else {
-                return true; // self is empty
+        if let Some(self_min) = self.first()
+            && let Some(self_max) = self.last()
+            && let Some(other_min) = other.first()
+            && let Some(other_max) = other.last()
+        {
+            let mut self_iter = self.iter();
+            match self_min.cmp(other_min) {
+                Less => return false,
+                Equal => self_iter.next(),
+                Greater => None,
             };
-        let (other_min, other_max) =
-            if let (Some(other_min), Some(other_max)) = (other.first(), other.last()) {
-                (other_min, other_max)
-            } else {
-                return false; // other is empty
+            match self_max.cmp(other_max) {
+                Greater => return false,
+                Equal => self_iter.next_back(),
+                Less => None,
             };
```

Comment:
Since these values are never used, I find the `None` arms very confusing.

Reply:
Done.